packages/ts-api-react/src/useApiCall.ts

```diff
@@ -1,13 +1,64 @@
-import { ApiEndpointProps } from "@thunderstore/thunderstore-api";
+import {
+  ApiEndpointProps,
+  MapUserFacingErrorOptions,
+  mapApiErrorToUserFacingError,
+} from "@thunderstore/thunderstore-api";
 
+/**
+ * Describes an API endpoint invocation with typed params and data payload.
+ */
 export type ApiEndpoint<Params, QueryParams, Data, Return> = (
   props: ApiEndpointProps<Params, QueryParams, Data>
 ) => Return;
 
+/**
+ * Options that control automatic error mapping for `useApiCall`.
+ */
+export type UseApiCallOptions = {
+  mapErrors?: boolean;
+  errorOptions?: MapUserFacingErrorOptions;
+};
+
+/**
+ * Wraps API endpoints to optionally map thrown errors into `UserFacingError` instances.
+ */
 export function useApiCall<Params, QueryParams, Data, Return>(
-  endpoint: ApiEndpoint<Params, QueryParams, Data, Return>
+  endpoint: ApiEndpoint<Params, QueryParams, Data, Return>,
+  options: UseApiCallOptions = {}
 ): (props: ApiEndpointProps<Params, QueryParams, Data>) => Return {
+  const shouldMapErrors = options.mapErrors ?? true;
+
   return (props: ApiEndpointProps<Params, QueryParams, Data>) => {
-    return endpoint(props);
+    try {
+      const result = endpoint(props);
+
+      if (!shouldMapErrors) {
+        return result;
+      }
+
+      if (isPromise(result)) {
```

Comment:
Minor: would be a bit clearer if these two if-conditions were combined ( `if (shouldMapErrors && isPromise(result)) {` )